packages/suite-desktop/src-electron/modules/index.ts

```diff
@@ -2,7 +2,39 @@
 /* eslint-disable global-require */
```

Comment:
can we rename this to something else that index? I would like to use less index files as possible, because current state is really confusing and has no clear rules:
1. When should I always it use for reexport other things if folder?
2. Should I put some logic or not here?
3. When should I put index to folder and when not?

All these things are super hard to define and best solution is just not to use index at all. Also indexes slowing things like HMR incremental recompilation and causing hidden circ deps etc. I suggess using name for this file just `modules` or `moduleLoader`

Reply:
i think in this case index does make sense. this is the only file from "modules" directory which is imported (in app.ts) the rest is dynamically required by this script.
I get your point in case of `suite` package but in case of `src-electron` there is no hot reloading, it's nodejs env

Reply:
using some custom name here may be even more confusing, which of the file is a module and which is not?

Reply:
I agree it makes sense here, but in general it is just another rule to list, how do I recognize where it makes sense and where not? Which packages are ok and which not? This could be really confusing for someone without your knowledge.

I still think `moduleLoader` is good name that is really good description on what this file do. Or maybe move this file out of this folder and leave just real "modules" here to be absolutely clear?

IDK, this is not a big issue for me I will leave it up to you, feel free to resolve if you want to keep index.

Reply:
`moduleLoader` is not a good name because other modules are using `module-loader` convention, did you check this directory and what this file does?

Reply:
I can see function that takes some array of modules and then load them? That is something I would call module loader. Other files seems to me like definition of modules but I don't see anything about loading modules in these files. I am not sure about that, maybe there is something what I missed.

Sorry for bothering you, I am just trying to understand how this works.

Reply:
Sure thing. Basically all modules from this directory except of the `index` (or `module-loader`) are dynamically required depending on build env (dev/prod) They should not imported anywhere else, no other file than `index` got reference to them (technically even `index` is not importing them directly using `import`)

`index` on the other hand is different, it's imported by `app.ts` and it purpose is to "indexing the content" of the whole "modules" directory providing/loading only specified modules to `app.ts`, this is the only "entrypoint" to modules.
If i rename it to `module-loader` it will get lost between other files, there will be no indication that this particular file is "special"

Reply:
and why i'm moving this? because of architectural mess.
All `modules` are using relation `module >  lib` (example: tor module is implementing Tor lib). 
Until now (current develop) `module` was the only lib with reversed relation `lib > load modules` and this was giving me a headache every time when it cross my path (lib named "module" > list of modules defined in "config" > loading files in opposite direction)

Reply:
Thanks for explanation, that makes sense and it's much better. If you think about "index.ts" like it something that indexes modules then name make sense, but still it could be quite suprising for someone who is used to that index should be just entry point with reexport. But I think since this is very special case so we can keep it.

Reply:
that's correct. this is an entry point with reexport but with few extra steps :) (dynamic conditional reexport)